frontend/src/metabase/home/components/HomeXraySection/HomeXraySection.tsx

```diff
@@ -33,10 +31,10 @@ import {
 export const HomeXraySection = () => {
   const databaseListState = useDatabaseListQuery();
   const database = getXrayDatabase(databaseListState.data);
-  const candidateListState = useDatabaseCandidateListQuery({
-    query: database ? { id: database.id } : undefined,
-    enabled: database != null,
-  });
+  const candidateListState = useListDatabaseCandidatesQuery(
+    { id: database?.id },
+    { skip: !database },
+  );
```

Comment:
@sloansparger question, if you have a better idea of how to deal with this. To preserve the conditional query here I had to loosen up the id type to accept undefined, even though that's actually not a valid API call. Any ideas about how to keep the type tighter but still allow an invalid value on skip?

Reply:
I don't know off the top of my head, but happy to help you look into it!

Reply:
https://redux-toolkit.js.org/rtk-query/usage-with-typescript#skipping-queries-with-typescript-using-skiptoken

```suggestion
  const candidateListState = useListDatabaseCandidatesQuery(database ? { id: database.id } : skipToken);
```

Reply:
nice! thanks @kamilmielnik